lib/matplotlib/axes/_axes.py

```diff
@@ -2604,6 +2605,137 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, captions=[], fmt="%g", mode="edge",
+                  padding=None, shifting=0, autoscale=True, **kwargs):
+        """
+        Label a bar plot.
+
+        Adds labels to bars in the given `.BarContainer`.
+
+        Parameters
+        ----------
+        container : `.BarContainer`
+            Container with all the bars and optionally errorbars.
+
+        captions : array-like, optional
+            A list of label texts, that should be displayed. If not given, the
+            label texts will be the data values formatted with *fmt*.
+
+        fmt : str, optional
+            A format string for the label. Default is '%g'
+
+        mode : {'edge', 'center'}, optional, default: 'edge'
+            Position of the label relative to the bar:
+
+            - 'edge': Placed at edge, the cumulative values will be shown.
+            - 'center': Placed at center, the individual values will be shown.
+
+        padding : float, optional
+            Space in points for label to leave the edge of the bar.
+
+        shifting : float, optional
+            Translation in points for label to leave the center of the bar.
+
+        autoscale : bool, optional
+            If ``True``, try to rescale to fit the labels. Default is ``True``.
+
+        Returns
+        -------
+        annotations
+            A list of `.Text` instances for the labels.
+        """
+        def nonefill(a, n):
+            return list(a) + [None] * (n - len(a))
+
+        def sign(x):
+            return 1 if x >= 0 else -1
+
+        cbook._check_in_list(['edge', 'center'], mode=mode)
+        if mode == "edge":
+            padding = padding or 3
+        elif mode == "center":
+            padding = padding or 0
```

Comment:
Can be shortened to
~~~
        if padding is None:
            padding = 3 if mode == "edge" else 0
~~~

Also `padding = padding or 3` is actually dangerous because people might try to pass `padding=False`, which will silently be converted to 3. In the modified version, at least the code will fail later on (errors should never pass silently).